test/integration/generated/datafactory/src/operations/triggers.ts

```diff
@@ -271,10 +271,12 @@ export class TriggersImpl implements Triggers {
       { resourceGroupName, factoryName, triggerName, options },
       subscribeToEventsOperationSpec
     );
-    return new LroEngine(lro, {
+    const poller = new LroEngine(lro, {
       resumeFrom: options?.resumeFrom,
       intervalInMs: options?.updateIntervalInMs
     });
+    await poller.poll();
```

Comment:
@deyaaeldeen, this change is related to the discussion about sending the initial request for LRO when creating the poller.

I believe this behavior change should be in `@azure/core-lro` either by default or based on an option, rather manually calling `.poll()` in the generated code. What do you think?

Reply:
@joheredi `LroEngine` is a class and it is not possible to make the constructor send requests because it cannot return a promise. The other thing we could do is to make core-lro export a factory function that creates an instance of the engine and activates it but that will require some discussion with @bterlson and @xirzec.

I am happy to explore this further but for now we can go with the current change until we decide on/create the factory.

Reply:
eventually, in core-lro@v3, I would like to move away from the class and just have a function to create the poller, this will make it easier to create an activated poller.

Reply:
Thanks for the clarification @deyaaeldeen. I'm fine with the change then